### PR TITLE
feat(terraform-version): implement tfenv version file updates

### LINKED ISSUE #7749: Support tfenv
**What would you like Renovate to be able to do?**

Support the `tfenv` manager

**Did you already have any implementation ideas?**

Similar to `nvm` and others. Reference: https://github.com/tfutils/tfenv#terraform-version-file

### PR DESCRIPTION
**Pull request by secustor**
## Changes:   Implement tfenv manager 

Closes #7749

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added unit tests, or
- [x] Unit tests + ran on a real repository

https://github.com/secustor/terraform-version

### CODE REVIEW

**Review by viceice — DISMISSED** [review]
Looks pretty easy, so LGTM

**Comment by secustor** [comment]
It isn't working a 100% correctly at the moment. 
Version in style of `0.13.0` get replaced by `v0.13.0` as the later is the versioning used by the Github releases of Terraform.    

I have posted a question regarding this on Slack.    
Do you have maybe an idea how to prevent this from happening?   
Maybe using default_config and a newVersion template, if this is supported?

**Comment by viceice** [comment]
If the file needs the `v` prefix removed, we can use `extractVersion` config to remove it.

**Comment by secustor** [comment]
Thanks that works

File: lib/manager/terraform-version/index.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import * as hashicorpVersioning from '../../versioning/hashicorp';
+
+export { extractPackageFile } from './extract';
+
+export const defaultConfig = {
+  fileMatch: ['(^|/)\\.terraform-version$'],
+  versioning: hashicorpVersioning.id,
+  extractVersion: '^v(?<version>.*)$',
```
Comment (viceice) [inline lib/manager/terraform-version/index.ts:8]:
```suggestion
  extractVersion: '^v?(?<version>.*)$',
```

What about suportion the prefix as optional?

Comment (secustor) [reply lib/manager/terraform-version/index.ts:8]: Terraform only uses `v` prefixed releases and as this repo is as default hardcoded, it makes from my point of view sense to follow that syntax.    
https://github.com/hashicorp/terraform/tags   

If someone want to overwrite this he can do this anytime.

**Review by rarkins — APPROVED** [review]

**Pull request merged**